pkg/reconciler/volumeclaim/pvchandler.go

```diff
@@ -51,50 +50,52 @@ func NewPVCHandler(clientset clientset.Interface, logger *zap.SugaredLogger) Pvc
-func (c *defaultPVCHandler) CreatePVCsForWorkspaces(ctx context.Context, wb []v1.WorkspaceBinding, ownerReference metav1.OwnerReference, namespace string) error {
-	var errs []error
-	for _, claim := range getPVCsWithoutAffinityAssistant(wb, ownerReference, namespace) {
-		_, err := c.clientset.CoreV1().PersistentVolumeClaims(claim.Namespace).Get(ctx, claim.Name, metav1.GetOptions{})
-		switch {
-		case apierrors.IsNotFound(err):
-			_, err := c.clientset.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(ctx, claim, metav1.CreateOptions{})
-			if err != nil && !apierrors.IsAlreadyExists(err) {
-				errs = append(errs, fmt.Errorf("failed to create PVC %s: %w", claim.Name, err))
-			}
-
-			if apierrors.IsAlreadyExists(err) {
-				c.logger.Infof("Tried to create PersistentVolumeClaim %s in namespace %s, but it already exists",
-					claim.Name, claim.Namespace)
-			}
-
-			if err == nil {
-				c.logger.Infof("Created PersistentVolumeClaim %s in namespace %s", claim.Name, claim.Namespace)
-			}
-		case err != nil:
-			errs = append(errs, fmt.Errorf("failed to retrieve PVC %s: %w", claim.Name, err))
-		}
+func (c *defaultPVCHandler) CreatePVCFromVolumeClaimTemplate(ctx context.Context, wb v1.WorkspaceBinding, ownerReference metav1.OwnerReference, namespace string) error {
+	claim := c.getPVCFromVolumeClaimTemplate(wb, ownerReference, namespace)
+	if claim == nil {
+		return nil
 	}
-	return errorutils.NewAggregate(errs)
-}
 
-func getPVCsWithoutAffinityAssistant(workspaceBindings []v1.WorkspaceBinding, ownerReference metav1.OwnerReference, namespace string) map[string]*corev1.PersistentVolumeClaim {
-	claims := make(map[string]*corev1.PersistentVolumeClaim)
-	for _, workspaceBinding := range workspaceBindings {
-		if workspaceBinding.VolumeClaimTemplate == nil {
-			continue
+	_, err := c.clientset.CoreV1().PersistentVolumeClaims(claim.Namespace).Get(ctx, claim.Name, metav1.GetOptions{})
+	switch {
+	case apierrors.IsNotFound(err):
+		_, err := c.clientset.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(ctx, claim, metav1.CreateOptions{})
+		if err != nil && !apierrors.IsAlreadyExists(err) {
+			return fmt.Errorf("failed to create PVC %s: %w", claim.Name, err)
+		}
+
+		if apierrors.IsAlreadyExists(err) {
+			c.logger.Infof("Tried to create PersistentVolumeClaim %s in namespace %s, but it already exists",
+				claim.Name, claim.Namespace)
 		}
 
-		claim := workspaceBinding.VolumeClaimTemplate.DeepCopy()
-		claim.Name = GetPVCNameWithoutAffinityAssistant(workspaceBinding.VolumeClaimTemplate.Name, workspaceBinding, ownerReference)
-		claim.Namespace = namespace
-		claim.OwnerReferences = []metav1.OwnerReference{ownerReference}
-		claims[workspaceBinding.Name] = claim
+		if err == nil {
+			c.logger.Infof("Created PersistentVolumeClaim %s in namespace %s", claim.Name, claim.Namespace)
+		}
+	case err != nil:
+		return fmt.Errorf("failed to retrieve PVC %s: %w", claim.Name, err)
 	}
```

Comment:
can this conditional logic be simplified?

Reply:
I refactored this part a bit. It is not really simplified but I found it is easier to read for me